README.md

```diff
@@ -120,26 +130,183 @@ A JRE is needed to run the code generators based on openapi-generator.
 
 ### Create new REST application
-To create a new REST application, run `harness-application` from the root of your solution.
+
+`harness-application` is a command-line tool used to create new applications based on predefined code templates. It allows users to quickly scaffold applications with backend, frontend, and database configurations.
+
+#### harness-application Usage
+
+```sh
+harness-application [name] [-t TEMPLATE]
+```
+
+#### harness-application Arguments
+
+- `name` *(required)* – The name of the application to be created.
+
+#### harness-application Options
+
+- `-h, --help` – Displays the help message and exits.
+- `-t TEMPLATES, --template TEMPLATES` – Specifies one or more templates to use when creating the application.
+
+#### Available Templates
+
+The following templates can be used with the `-t` flag:
+
+- **flask-server** – Backend Flask server based on OpenAPI.
+- **webapp** – Full-stack React web application with both frontend and backend.
+- **db-postgres** – PostgreSQL database setup.
+- **db-neo4j** – Neo4j database setup.
+- **db-mongo** – MongoDB database setup.
+- **django-fastapi** – FastAPI and Django backend based on OpenAPI.
+- **django-ninja** – Django Ninja backend.
+
+#### harness-application Examples
+
+##### Create a New Flask-Based Microservice Application
+
+```sh
+harness-application myapp
+```
+
+##### Create a Full-Stack Web Application
+
+```sh
+harness-application myapp -t webapp
+```
+
+##### Create a Web Application with Mongo Database
+
+```sh
+harness-application myapp -t webapp -t db-mongo
+```
+
+##### Display Help Information
+
+```sh
+harness-application --help
+```
+
+#### harness-application Notes
+
+- Multiple templates can be specified concatenating the -t parameter.
+- The tool generates the necessary scaffolding for the chosen templates.
+- Ensure you have the required dependencies installed before running the generated application.
+- For more information, run `harness-application --help` or check out the additional documentation:
+  - [Applications README](./docs/applications/README.md)
+  - [Developer Guide](./docs/dev.md)
 
 ### Generate server and client code from openapi
-To (re)generate the code for your applications, run `harness-generate` from the root.
-The script will look for all openapi applications, and regenerate the Flask server code and documentation.
-Note: the script will eventually override any manually modified file. To avoid that, define a file openapi-generator-ignore.
+
+To (re)generate the code for your applications, run `harness-generate`.
+`harness-generate` is a command-line tool used to generate client code, server stubs, and model libraries for applications. It walks through the filesystem inside the `./applications` folder to create and update application scaffolding. The tool supports different generation modes and allows for both interactive and non-interactive usage.
+
+#### Usage
```

Comment:
It looks a lot of details for the main readme, what about moving to docs? There is also this one: https://github.com/MetaCell/cloud-harness/blob/2d03fd84b0532274c05e1aee104edf32ecb91e94/docs/applications/harness-application.md. Maybe we can create docs/code-generation.md and merge this section and that file?